test/functional/wallet_analyzerawtx.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test analyzerawtransaction.
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_approx,
+    assert_equal,
+)
+
+class AnalyzeTxTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def setup_network(self, split=False):
+        self.setup_nodes()
+
+    def run_test(self):
+        node0 = self.nodes[0]
+        node1 = self.nodes[1]
+        self.connect_nodes(0, 1)
+
+        node0.generate(1) # Leave IBD
+
+        node0.createwallet(wallet_name='w0')
+        w0 = node0.get_wallet_rpc('w0')
+        node1.createwallet(wallet_name='w1')
+        w1 = node1.get_wallet_rpc('w1')
+
+        node0.generatetoaddress(101, w0.getnewaddress())
+        assert_equal(w0.getbalance(), 50.0)
+        assert_equal(w1.getbalance(), 0.0)
+
+        address1 = w1.getnewaddress()
+
+        tx = node1.createrawtransaction([], [{address1: 5.0}])
+
+        # node0 should be unaffected
+        assert_equal(w0.analyzerawtransaction(tx), 0.0)
+
+        # node1 should see +5 balance
+        assert_equal(w1.analyzerawtransaction(tx), 5.0)
+
+        # w0 funds transaction; it should now see a decrease in (tx fee and payment), and w1 should see the same as above
+        funding = w0.fundrawtransaction(tx)
+        tx = funding["hex"]
+        bitcoin_fee = float(funding["fee"])
+
+        # node0 sees fee + 5 btc decrease
+        assert_approx(w0.analyzerawtransaction(tx), -(5.0 + bitcoin_fee))
+
+        # node1 sees same as before
+        analysis = w1.analyzerawtransaction(tx)
```

Comment:
`test/functional/wallet_analyzerawtx.py:60:9: F841 local variable 'analysis' is assigned to but never used`

Reply:
Thanks, fixed.